### PR TITLE
Monitoring and Analytics dashboard

### PR DESCRIPTION
**Pull request by cindylovescoding**
1. Add tabs for Monitoring and Analytics dashboard
2. Add onboarding flow for Monitoring and Analytics system invoker
3. Route system invoker call to runtime host backend

### CODE REVIEW

**Comment by stevenernst131** [comment]
@cindylovescoding Are there any screenshots?

**Comment by cindylovescoding** [comment]
The csx files for monitoring and analytics dashboard developing are still in progress, so the content of these two dashboards may not look like the screenshots eventually.

Monitoring Dashboard:

image

Analytics Dashboard: 

image

**Comment by cindylovescoding** [comment]
Yes there are. Let me add some:)

File: AngularApp/src/app/modules/dashboard/services/applens-diagnostic.service.ts [hunk 1]
```diff
@@ -26,11 +37,23 @@ export class ApplensDiagnosticService {
...
   }
 
-  getCompilerResponse(body: any): Observable<QueryResponse<DetectorResponse>> {
+  getCompilerResponse(body: any, isSystemInvoker: boolean, detectorId: string = '', dataSource: string = '', timeRange: string = ''): Observable<QueryResponse<DetectorResponse>> {
     body.resource = this._resourceService.getRequestBody();
-    return this._diagnosticApi.getCompilerResponse(
-      this._resourceService.versionPrefix,
-      this._resourceService.getCurrentResourceId(true),
-      body);
+    if (isSystemInvoker == false)
```
Comment (stevenernst131) [inline AngularApp/src/app/modules/dashboard/services/applens-diagnostic.service.ts]: either 

```
!isSystemInvoker

or

isSystemInvoker === false
```

File: AngularApp/src/app/modules/dashboard/tabs/tab-analytics-develop/tab-analytics-develop.component.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+@Component({
+  selector: 'tab-analytics-develop',
+  templateUrl: './tab-analytics-develop.component.html',
+  styleUrls: ['./tab-analytics-develop.component.css']
+})
+export class TabAnalyticsDevelopComponent implements OnInit {
+
+  DevelopMode = DevelopMode;
+  private detectorId: string;
+  private dataSourceMapping: Map<string, string> = new Map<string, string>([
```
Comment (stevenernst131) [inline AngularApp/src/app/modules/dashboard/tabs/tab-analytics-develop/tab-analytics-develop.component.ts:15]: There seems to be a lot in common between the analytics tab and the monitoring tab. For example, the drop downs for all, applens, portal, and the time selector. Seems like this is something that could be broken out into a single component.

File: AngularApp/src/app/modules/dashboard/dashboard.module.ts [hunk 3]
```diff
@@ -76,6 +80,26 @@ export const DashboardModuleRoutes: ModuleWithProviders = RouterModule.forChild(
...
+            component: TabMonitoringComponent
+          }          
+          ,
+          {
+            path: 'analytics',
+            component: TabAnalyticsDashboardComponent
+          }
+          ,
+          {
+            path: 'statisticsQuery/monitoring',
```
Comment (stevenernst131) [inline AngularApp/src/app/modules/dashboard/dashboard.module.ts]: should we follow the pattern of adding a '/edit' to the end of the url for the edit view?

**Review by stevenernst131 — APPROVED** [review]

File: AngularApp/src/app/modules/dashboard/dashboard.module.ts (hunk 3 above)
Comment (cindylovescoding) [reply AngularApp/src/app/modules/dashboard/dashboard.module.ts]: Make sense:) Let me change that.

File: AngularApp/src/app/modules/dashboard/services/applens-diagnostic.service.ts (hunk 1 above)
Comment (cindylovescoding) [reply AngularApp/src/app/modules/dashboard/services/applens-diagnostic.service.ts]: Oh I see.. Fixed this.

File: AngularApp/src/app/modules/dashboard/tabs/tab-analytics-develop/tab-analytics-develop.component.ts (hunk 2 above)
Comment (cindylovescoding) [reply AngularApp/src/app/modules/dashboard/tabs/tab-analytics-develop/tab-analytics-develop.component.ts:15]: Yes currently analytics/monitoring dashboard tab are sharing components, and analytics/monitoring developing tabs are sharing components. Totally  agree there will be something I can retrieve as a common component. Also I need some layout change for this design. Will put that layout change together with this refactor work.

**Pull request merged**